src/bin/zmimapdctl

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+# 
+# ***** BEGIN LICENSE BLOCK *****
+# Zimbra Collaboration Suite Server
+# Copyright (C) 2017 Synacor, Inc.
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software Foundation,
+# version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this program.
+# If not, see <https://www.gnu.org/licenses/>.
+# ***** END LICENSE BLOCK *****
+# 
+
+# Fix shellcheck warning.  This value is updated by zmsetvars
+zimbra_java_home=
+
+SDIR=$(dirname "$0")
+source "${SDIR}/zmshutil" || exit 1
+zmsetvars
+
+if [ ! -d "${zimbra_java_home}" ]; then
+  echo "Did not find java at '${jar_prefix}'" >&2
+  exit 1
+fi
+
+#
+# TODO-JVM options
+# 
+
+PID_FILE=/opt/zimbra/log/imapdaemon.pid
+NOHUP_FILE=/opt/zimbra/log/imapdaemon.out
+
+CAT=/bin/cat
+JAVA="${zimbra_java_home}/bin/java"
+LS=/bin/ls
+KILL=/bin/kill
+NOHUP=/usr/bin/nohup
+PS=/bin/ps
+RM=/bin/rm
+
+#
+# Utility
+#
+
+function is_running {
+  if [ -f $PID_FILE ] && $PS -p "$($CAT $PID_FILE)" >/dev/null; then
+    echo yes
+  else
+    echo no
+  fi
+}
+
+#
+# Main
+#
+
+case "$1" in
+  start)
+    if [ "$(is_running)" = "no" ]; then
+      /opt/zimbra/bin/zmlocalconfig -e imap_always_use_remote_store=true
+      ${NOHUP} "${JAVA}" -cp /opt/zimbra/lib/jars/*: com.zimbra.cs.imap.ImapDaemon 1>$NOHUP_FILE 2>&1 &
+      PID=$!
+      echo -n $PID > $PID_FILE
+      # TODO - find deterministic way to tell when Java has completed startup
```

Comment:
Will there be a Jira for this if not doing it as part of this change?  I think the "sleep 2" is too brittle.  Would prefer for us not to ship without this addressed.